lib/listener/mac/ITunesListener.cpp

```diff
@@ -228,19 +248,30 @@ ITunesListener::callback( CFDictionaryRef info )
 
 }
 
+QString //static
+ITunesListener::getPlayerAppId()
+{
+    const char* code = 
+    "try\n"
+        "tell me to get application id \"com.apple.Music\"\n"
+        "return \"com.apple.Music\"\n"
+    "on error\n"
+        "return \"com.apple.iTunes\"\n"
+    "end try\n";
+    return AppleScript( code ).exec();
+}
 
 bool //static
-ITunesListener::iTunesIsPlaying()
+ITunesListener::isPlaying()
 {
-    const char* code = "tell application \"iTunes\" to if running then return player state is playing";
+    QString code = "tell application id \"" + m_playerAppId + "\" to if running then return player state is playing";
```

Comment:
I notice this is now a QString? Why?

Reply:
You can't concat `const char* `

Reply:
And the AppleScript() function definition actually accepts a `QString`, not `const char*`